datafusion/expr/src/udaf.rs

```diff
@@ -262,6 +262,19 @@ impl AggregateUDF {
         self.inner.is_descending()
     }
 
+    /// Returns true if the function is min. Used by the optimizer
+    pub fn is_min(&self) -> bool {
```

Comment:
This is not we should do... 
We need to understand the context and have a general function instead of a specialize name matching function.

`default_value` is a good example, it is only used in count for now, but it could extend to any function if needed.

Reply:
@jayzhan211 looking to this comment from @alamb  https://github.com/apache/datafusion/issues/11151#issuecomment-2198513514 it might be that those "general functions" might not exists, will do some homework

Reply:
If there is really no "general context", I think it is fine to just leave them as it is. Having special function matching in Impl Trait doesn't make sense to me 🤔

Reply:
I agree with @jayzhan211 that it is important that these methods describe some property about the function rather than "what the function is"

What if we added a function like this:

```rust
/// Return the value of this aggregate function from statistics
///
/// If the value of this aggregate, can be determined using only the 
/// statistics, return `Some(value)`, otherwise return `None` (the default)
///
/// # Arguments
/// * `statistics`: the statistics describing the input to this aggregate functions
/// * `args`: the arguments passed to the aggregate function 
///
/// The value of some aggregate functions such as `COUNT`, `MIN` and `MAX` 
/// can be determined using statistics, if known
///
fn value_from_stats(&self, statistics: &Statistics, arguments: &[Arc<PhysicalExpr>]) -> Option<ScalarValue> { None }

```

I think you could then implement this function for min / max and count (moving logic out of the aggregate statistics optimizer). It might need some other information (like schema for types, for example) but I think it would be pretty straight forward